### DISCUSSION TITLE
2025 Phoenix Beta

### DISCUSSION THREAD

**Discussion by daltzctr** [wpilibsuite/2025Beta#7] [General]
We are excited to announce our first beta release for 2025. This release contains many changes, improvements and enhancements for 2025. 

We are excited to read any feedback the community may have.

> [!NOTE]  
> Users should use "Phoenix Tuner X Preview" ([link](
https://apps.microsoft.com/store/detail/9N17NS6NM06V)) (a separate application in the Windows Store) for usage with beta 1

# Download

Links for the vendordeps, a detailed changelog, and more can be found below:

https://github.com/CrossTheRoadElec/Phoenix-Releases/releases/tag/v25.0.0-beta-1

**Comment by ArchdukeTim** [comment]
Is there still a plan to support running multiple motors with a single tuner x application for 2025?

**Reply by daltzctr** [reply]
This is still an in-development effort. We have already made architectural improvements in the diagnostic server to enable this, but UI and Tuner development is in-progress.

**Comment by daltzctr** [comment]
Beta 2 is available now https://github.com/CrossTheRoadElec/Phoenix-Releases/releases/tag/v25.0.0-beta-2! See the [changelog](https://api.ctr-electronics.com/changelog) for a detailed list of changes and subscribe to our [RSS](https://api.ctr-electronics.com/rss/rss.xml) feed for notifications when new releases post.

**Comment by ArchdukeTim** [comment]
Various signal logging seems to have broken with the latest update

**Reply by daltzctr** [reply]
Could you elaborate on what you mean?

**Reply by ArchdukeTim** [reply]
Well it seemed like all signals were broken. I figured it was probably a tuner x out of date issue but the windows store didn't say an update was available. Closing and reopening I can confirm I'm _now_ on 2025.1.0.0, and things are working again. I suspect that was the issue. 

When I first opened tuner x it knew there was an updated firmware for the talon though. Not sure if that is built into the app or not

**Reply by daltzctr** [reply]
I see the problem now. We did back break owlet/logging from beta 1 to beta 2. Tuner 2025.1.1.0 is on it's way to the stores and will have a fix (and will also make it work on various logs for various api levels as well).

**Comment by daltzctr** [comment]
Hey folks. Just a couple of announcements.

- [CANrange](https://store.ctr-electronics.com/products/canrange) is in-stock and shipping now!
- [Phoenix Beta 4](https://github.com/CrossTheRoadElec/Phoenix-Releases/releases/tag/v25.0.0-beta-4) is available now with support for CANrange hardware and full simulation support.

**Comment by viggy96** [comment]
Does `applyConfigs()` only apply the configs that have changed, or apply everything again?

**Reply by bhall-ctre** [reply]
When you apply a full `<Device>Configuration` object, such as `TalonFXConfiguration`, then we apply everything if the [`FutureProofConfigs`](https://api.ctr-electronics.com/phoenix6/latest/java/com/ctre/phoenix6/configs/TalonFXConfiguration.html#FutureProofConfigs) field is true (default). If that field is false or you're applying one of the config groups, such as `Slot0Configs`, then we only apply configs that have changed since the last call to `device.getConfigurator().apply(...)`.
